packages/lodestar/src/network/encoders/response.ts

```diff
@@ -0,0 +1,107 @@
+import {IResponseChunk} from "./interface";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {ILogger} from "@chainsafe/lodestar-utils/lib/logger";
+import {Method, MethodResponseType, Methods, ReqRespEncoding, RpcResponseStatus} from "../../constants";
+import {decode, encode} from "varint";
+import {encodeResponseStatus, getCompressor, getDecompressor} from "./utils";
+import BufferList from "bl";
+import {ResponseBody} from "@chainsafe/lodestar-types";
+
+export function eth2ResponseEncode(
+  config: IBeaconConfig, logger: ILogger, method: Method, encoding: ReqRespEncoding
+): (source: AsyncIterable<IResponseChunk>) => AsyncIterable<Buffer> {
+  return (source => {
+    return (async function*() {
+      const type = Methods[method].responseSSZType(config);
+      let compressor = getCompressor(encoding);
+      if(!type) {
+        return;
+      }
+      for await (const chunk of source) {
```

Comment:
question: why do we yield status, ssz length and compressed data separately instead of just concatenate them in a single buffer and yield it like the previous version?

Reply:
Because recipient needs status and length so it knows how much frames it needs to decompress. with compression we are sending compressed data frame by frame while previously we would just send one blob of data so we didn't really need length prefix anyways